src/main/java/com/google/step/snippet/servlets/SearchServlet.java

```diff
@@ -74,8 +77,9 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
      */
     String w3Link = getLink(W3_CSE_ID, query);
     if (w3Link != null) {
-      allLinks.add(w3Link);
-      // TODO: Call scraping function to return JSON card content
+      W3SchoolClient client = new W3SchoolClient();
+      Card w3Card = client.search(w3Link);
+      allCards.add(w3Card);
```

Comment:
Need to add null check here

`if (w3Card != null) {
allCards.add(w3Card);
}`